### PR TITLE
Contour 1.0

### PR DESCRIPTION
**Pull request by dulltz**
- Upgrade contour to v1.0
- Use `HTTPProxy` instead of `IngressRoute`
- Create the OTS patch
  - Note that some manifests does not inherit the upstream

### CODE REVIEW

File: ingress/base/contour/deployment.yaml [hunk 1]
```diff
@@ -4,127 +4,105 @@ metadata:
   labels:
     app.kubernetes.io/name: contour
   name: contour
-  namespace: ingress
+  namespace: projectcontour
 spec:
+  strategy:
```
Comment (ymmt2005) [inline ingress/base/contour/deployment.yaml]: seems an invalid YAML?

File: ingress/base/contour/deployment.yaml [hunk 2]
```diff
@@ -4,127 +4,105 @@ metadata:
...
   selector:
     matchLabels:
       app.kubernetes.io/name: contour
-  replicas: 2
```
Comment (ymmt2005) [inline ingress/base/contour/deployment.yaml]: Why do you remove this?

File: ingress/base/contour/contour-config.yaml [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-    # disablePermitInsecure: false
+    disablePermitInsecure: false
     tls:
-      # minimum TLS version that Contour will negotiate
       minimumProtocolVersion: "1.2"
+    leaderelection:
+      configmap-name: contour-leader
+      configmap-namespace: ingress
+    accesslog-format: envoy
+
```
Comment (ymmt2005) [inline ingress/base/contour/contour-config.yaml]: garbage line

File: ingress/base/contour/deployment.yaml (hunk 1 above)
Comment (ymmt2005) [inline ingress/base/contour/deployment.yaml]: The off-the-shelf strategy seems good.

**Comment by ymmt2005** [comment]
Please change the PR so that it does not patch deployments/daemonsets but instead
replace the whole manifests of deployments/daemonsets entirely with ours.

File: ingress/base/contour/deployment.yaml (hunk 1 above)
Comment (kfyharukz) [reply ingress/base/contour/deployment.yaml]: I wrote the same strategy in the upstream.
Also, I added the strategy to envoy deployment (it is different from the upstream, but I think it should be written.)

File: docs/maintenance.md [hunk 4]
```diff
@@ -97,6 +97,9 @@ cd contour
 git diff vA.B.C...vX.Y.Z examples/contour
 ```
 
+Note that not all manifests inherit the upstream (e.g. contour and envoy deployments).
+Please check `kustomization.yaml` which manifest inherits or not.
+
```
Comment (ymmt2005) [inline docs/maintenance.md]: Please add these descriptions:

- We do not use contour's certificate issuance feature, but use cert-manager to issue certificates required for gRPC.
- We change Envoy from DaemonSet to Deployment.

Comment (kfyharukz) [reply docs/maintenance.md]: Thanks your comments. I reflected them.

**Pull request merged**